### ISSUE TITLE
Include tools needed by rust-mos build

### ISSUE THREAD

**Issue by mrk-its** [llvm-mos/llvm-mos#275]
To build rust-mos following changes to MOS.cmake are required:

```
--- clang/cmake/caches/MOS.cmake        2022-12-29 22:57:38.304883166 +0100
+++ clang/cmake/caches/RUST-MOS.cmake   2022-12-30 00:36:54.994806413 +0100
@@ -3,7 +3,7 @@
 # Usage for configuring:
 #   cmake -C [path-to-this-file] ...

-set(LLVM_TARGETS_TO_BUILD ""
+set(LLVM_TARGETS_TO_BUILD "X86"
     CACHE STRING "LLVM targets to build")
 set(LLVM_EXPERIMENTAL_TARGETS_TO_BUILD "MOS"
     CACHE STRING "LLVM experimental targets to build")
@@ -15,6 +15,8 @@
     CACHE BOOL "LLVM install toolchain only")

 set(LLVM_DEFAULT_TARGET_TRIPLE "mos-unknown-unknown" CACHE STRING "")
+set(LLVM_TOOLCHAIN_UTILITIES FileCheck CACHE STRING "")
+set(LLVM_INSTALL_UTILS "ON" CACHE STRING "")

 # The following options are principally to reduce space on Github action
 # runner builds. They make smaller, and possibly slower, releases; but the
@@ -55,6 +57,7 @@

 # Ship the release with these tools
 set(LLVM_MOS_TOOLCHAIN_TOOLS
+    llvm-config
     llvm-addr2line
     llvm-ar
     llvm-cxxfilt
```

The most significant change is adding X86 (host) target, I'm just investigating if it is really necessary. But even if not, host target is required for running rust build (buils.rs) scripts, which is really nice feature to have

**Comment by mrk-its** [comment]
It looks like `-DLLVM_INSTALL_TOOLCHAIN_ONLY=OFF` is also needed, without it rust-mos build failed on missing `include/llvm-c/BitReader.h` header

**Comment by mysterymath** [comment]
These changes LGTM, especially since we do set the default target triple explicitly. Could you make a PR with them when you get the chance? I'll merge it immediately.

**Comment by mrk-its** [comment]
> These changes LGTM, especially since we do set the default target triple explicitly. Could you make a PR with them when you get the chance? I'll merge it immediately.

Here it is: https://github.com/llvm-mos/llvm-mos/pull/276

Thanks!

**Comment by mysterymath** [comment]
Closed by #276.
